echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/artifacts/ArtifactMatcher.java

```diff
@@ -129,7 +130,14 @@ public static boolean isJsonPathConstraintInPayload(final Map constraints, final
   }
 
   private static boolean matches(String us, String other) {
-    return Pattern.compile(us).asPredicate().test(other);
+    Pattern p;
+    try {
+      p = Pattern.compile(us);
+    } catch (PatternSyntaxException ex) {
+      log.error("Invalid Regex pattern for constraint, will never match any payload: " + us);
```

Comment:
I bet there's some helpful information in the exception message.  I'd include it in the log message, or maybe the entire exception, so that the stack trace and any other cause shows up.

Reply:
Thanks!